crates/goose/src/providers/base.rs

```diff
@@ -338,6 +339,64 @@ pub trait Provider: Send + Sync {
             self.get_model_config().model_name
         }
     }
+
+    /// Generate a session name/description based on the conversation history
+    ///
+    /// This method can be overridden by providers to implement custom session naming strategies.
+    /// The default implementation creates a prompt asking for a concise description in 4 words or less.
+    ///
+    /// # Arguments
+    /// * `messages` - The conversation history as a sequence of messages
+    ///
+    /// # Returns
+    /// A string containing the session name/description, sanitized and truncated to be safe for use
+    ///
+    /// # Errors
+    /// ProviderError if the session name generation fails
+    async fn generate_session_name(&self, messages: &[Message]) -> Result<String, ProviderError> {
+        // Create a special message asking for a concise description
+        let mut description_prompt = "Based on the conversation so far, provide a concise description of this session in 4 words or less. This will be used for finding the session later in a UI with limited space - reply *ONLY* with the description".to_string();
+
+        // Get context from messages so far, limiting each message to 300 chars for security
+        let context: Vec<String> = messages
+            .iter()
+            .filter(|m| m.role == rmcp::model::Role::User)
+            .take(3) // Use up to first 3 user messages for context
+            .map(|m| {
+                let text = m.as_concat_text();
+                safe_truncate(&text, 300)
```

Comment:
I think we can drop this truncate. especially for sessions where the first message is really quite long because the user knows what they are doing, you probably don't want to do this. as for security, these messages have gone through the agent already. if we do want to keep this make it 3000 or so.

I'm aware this is pre-existing code, but still might as well make things better